### PR TITLE
rax_files_objects: refactoring

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
- simplifications
- use of comprehensions
- better use of exceptions
- improvements on the documentation blob
- src and dest mutually exclusive in module definition

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/cloud/rackspace/rax_files_objects.py

### CODE REVIEW

File: plugins/modules/cloud/rackspace/rax_files_objects.py [hunk 1]
```diff
@@ -319,8 +311,7 @@ def download(module, cf, container, src, dest, structure):
     # Accept a single object name or a comma-separated list of objs
     # If not specified, get the entire container
     if src:
-        objs = src.split(',')
-        objs = map(str.strip, objs)
+        objs = map(str.strip, objs.split(','))
```
Comment (felixfontein) [inline plugins/modules/cloud/rackspace/rax_files_objects.py:314]:
```suggestion
        objs = map(str.strip, src.split(','))
```

File: plugins/modules/cloud/rackspace/rax_files_objects.py [hunk 2]
```diff
@@ -582,7 +507,10 @@ def main():
 
     module = AnsibleModule(
         argument_spec=argument_spec,
-        required_together=rax_required_together()
+        required_together=rax_required_together(),
+        mutually_exclusive=[
+            ('src', 'dest'),
```
Comment (felixfontein) [inline plugins/modules/cloud/rackspace/rax_files_objects.py:512]: Right now they are only mutually exclusive for `method=delete` or `type=meta`. For `method=put` and `type=file`, both can be specified (for actual files and not directories)

Comment (russoz) [reply plugins/modules/cloud/rackspace/rax_files_objects.py]: okie, rolled back

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for improving this module!

**Pull request merged**